src/lib/operators/multi_predicate_join/multi_predicate_join_evaluator.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <vector>
+
+#include "operators/operator_join_predicate.hpp"
+#include "storage/segment_accessor.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class MultiPredicateJoinEvaluator {
+ public:
+  MultiPredicateJoinEvaluator(const Table& left, const Table& right,
+                              const std::vector<OperatorJoinPredicate>& join_predicates);
+
+  MultiPredicateJoinEvaluator(const MultiPredicateJoinEvaluator&) = default;
```

Comment:
Why have copy and move constructor? If not absolutely necessary, pls remove them

Reply:
In the HJ and SMJ we have an optional mult. pred. join evaluator. We emplace an instance of the evaluator if secondary join predicates exist:
```
if (!_secondary_join_predicates.empty()) {
      _multi_predicate_join_evaluator.emplace(MultiPredicateJoinEvaluator(*_sort_merge_join._input_left->get_output(),
                                                                          *_sort_merge_join.input_right()->get_output(),
                                                                          _secondary_join_predicates));
    }
```
So we do  need at least the move constructor to emplace the evaluator instance, don't we?

Reply:
emplace() uses the other constructor in line 14.

Reply:
I got it now. Changed the emplace call so the correct constructor is used and the move constructor can also be deleted. fb56cfb5da47244519aebbdaf85efb2810ba7fd7